webcompat/webhooks/helpers.py

```diff
@@ -17,20 +17,30 @@
 from webcompat.helpers import proxy_request
 
 
-def extract_browser_label(body):
-    """Parse the labels from the body in comment:
+def extract_metadata(body):
+    """Parse all the hidden comments for issue metadata
 
-    <!-- @browser: value -->.
-    Currently this only handles a single label,
-    because that's all that we set in webcompat.com.
+    <!-- @foo: bar -->.
+    Returns a dict with all such comments as members,
+    with foo, bar as key, value.
     """
-    match_list = re.search(r'<!--\s@(\w+):\s([^\d]+?)\s[\d\.]+\s-->', body)
-    if match_list:
-        # perhaps we do something more interesting depending on
-        # what groups(n)[0] is in the future.
-        # right now, match_list.groups(0) should look like:
-        # ('browser', 'firefox')
-        browser = match_list.groups(0)[1].lower()
+    match_list = re.findall(r'<!--\s@(\w+):\s(.+)\s-->', body)
+    # Build a dict but only for the first seen instance of a key
+    # (in case there are duplicates added by the user after)
+    metadata_dict = {}
```

Comment:
/me images Karl suggesting some cool dict comprehension to use here....

So, we can't just `dict(match_list)`, because if there were duplicate keys, the `dict()` constructor swallows them all except the last in the list. Which is the opposite of what we want, I think... 

https://github.com/webcompat/webcompat.com/blob/54e0d5d5a9f2bcfe143fa652894b4703591678c1/tests/test_webhook.py#L56

I don't recall if this was on purpose, but I think the logic here is in case the user somehow adds their own (incorrect) metadata comment, we are interested in the first that the app added. Which makes sense to me. Hence, this constructing the `metadata_dict` by hand.

Reply:
BTW, I double checked and `re.findall` will insert into the list from left to right, in order. So that's good.

Reply:
@miketaylr :) Ready for the fun?

```python
>>> foo = [('love', 'wonderful'), ('peace', 'awesome'), ('love', 'overrated')]
>>> foo
[('love', 'wonderful'), ('peace', 'awesome'), ('love', 'overrated')]
>>> dict(foo)
{'peace': 'awesome', 'love': 'overrated'}
>>> # Not what we want :) ok let's do a trick
>>> dict(reversed(foo))
{'peace': 'awesome', 'love': 'wonderful'}
```

So basically we reverse the order of the list so the last (once first) tuples will be the ones we keep. ;)

```python
metadata_dict = dict(reversed(match_list))
```